src/Drips.sol

```diff
@@ -557,14 +557,17 @@ library Drips {
                     _currTimestamp(),
                     newDefaultEnd
                 );
-                int256 amtPerSec = int256(uint256(currRecv.config.amtPerSec()));
-                // Move the start and end times if updated
-                _addDeltaRange(state, cycleSecs, currStart, newStart, -amtPerSec);
-                _addDeltaRange(state, cycleSecs, currEnd, newEnd, amtPerSec);
+                {
+                    int256 amtPerSec = int256(uint256(currRecv.config.amtPerSec()));
+                    // Move the start and end times if updated
+                    _addDeltaRange(state, cycleSecs, currStart, newStart, -amtPerSec);
+                    _addDeltaRange(state, cycleSecs, currEnd, newEnd, amtPerSec);
+                }
                 // Ensure that the user receives the updated cycles
-                uint32 startCycle = _cycleOf(newStart, cycleSecs);
-                if (state.nextReceivableCycle > startCycle) {
-                    state.nextReceivableCycle = startCycle;
+                uint32 currStartCycle = _cycleOf(currStart, cycleSecs);
+                uint32 newStartCycle = _cycleOf(newStart, cycleSecs);
+                if (currStartCycle > newStartCycle && state.nextReceivableCycle > newStartCycle) {
```

Comment:
It's an optimization, this PR doesn't fix any bugs and doesn't change the behavior of the contract.

You're absolutely right, the old logic of `state.nextReceivableCycle > newStartCycle` is enough. The trick here is that finding out that `currStartCycle <= newStartCycle` costs just a handful of gas but it implies that `state.nextReceivableCycle <= newStartCycle`, so we can skip access storage and save 2100 gas.

Reply:
Mhm, but then you would need a logical `OR` but this you can't do because `currStartCycle <= newStartCycle` is not enough in that case.

Because I still could have a second drip with an even earlier start date.

Reply:
This is enough, the goal is to ensure that `(state.nextReceivableCycle <= newStartCycle) == true`.

`&&` is always short-circuit, `if (currStartCycle > newStartCycle && state.nextReceivableCycle > newStartCycle)` is executed as
```
if(currStartCycle > newStartCycle) {
    if(state.nextReceivableCycle > newStartCycle) {
        ...
    }
}
```
If `currStartCycle <= newStartCycle`, then there's no point in reading `state.nextReceivableCycle`, it MUST have an already good value.

Reply:
Ah, Yes correct. I was focused on the case where you want to go into the IF condition but your optimization is when you can skip it.